### PR TITLE
Lock PHP version for CI workflows

### PR DESCRIPTION
**Pull request by ben-hamelin**
## Summary

Adds a github action and associated config that supports PHP setup for the CI workflow. Locks PHP at 7.3 to mimic the target environments.

## Metadata

| Question | Answer |
|----------|--------|
| Did you use a meaningful pull request title? | yes
| Did you apply meaningful labels to the pull request? | yes
| Did you perform a self review first? | yes
| Documentation reflects changes? | no
| `CHANGELOG` reflects changes? | yes
| Unit/Functional tests reflect changes? | n/a
| Did you perform browser testing? | yes
| Risk level | Low
| Relevant links | https://github.com/shivammathur/setup-php
https://stackoverflow.com/questions/57965588/how-to-test-with-different-versions-of-php-in-a-github-action

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,8 @@ modified Semantic Versioning scheme. See the "Versioning scheme" section of the
 
 ## [Unreleased]
 ### Added
+- RIG-177: Added the publisher module for publication nodes to go from hub to syndicated sites.
```
Comment (pfrilling) [inline CHANGELOG.md:13]: Rebase issue? This doesn't belong.

**Review by pfrilling — CHANGES_REQUESTED** [review]
Nitpick in the changelog. Otherwise, good to go.

File: CHANGELOG.md (hunk 1 above)
Comment (ben-hamelin) [reply CHANGELOG.md]: oops! Yep, got rid of it!

**Review by pfrilling — APPROVED** [review]
Looks good!

**Pull request merged**